### PR TITLE
Fixed issues with tenant administration (#640)

### LINKED ISSUE #640: Issues with tenant administration
- if I create a tenant with an URL prefix and later I want to edit it, I am unable to save the changes. The message "The url prefix is already used by another tenant" appears
- if I create a tenant with a host and a prefix and another tenant with a different host but the same prefix, I am unable to save changes to either of them without changing the prefix

### PR DESCRIPTION
**Pull request by lukaskabrt**
Fixes #640 

As described in the issue, original code required unique (host, prefix) pair when creating a tenant and unique prefix when editing a tenant. It also forbade user to save a tenant without changing the prefix.

The new code require unique (host, prefix) in both cases and fixes the problem with saving a tenant without changing the prefix.

### CODE REVIEW

**Comment by lukaskabrt** [comment]
Fixed additional issue with tenant URL in the Index.cshtml file - if a tenant host was defined, the invalid URL was generated (e.g. localhost/mydomain.com/urlprefix)

File: src/Orchard.Cms.Web/Modules/Orchard.Tenants/Extensions/UrlHelperExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Orchard.Environment.Shell;
+using System;
+using System.Linq;
+
+namespace Orchard.Tenants.Extensions
+{
+    public static class UrlHelperExtensions
+    {
+        public static string Tenant(this IUrlHelper urlHelper, ShellSettings tenantShellSettings)
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.Tenants/Extensions/UrlHelperExtensions.cs]: Red alert!
Don't have time to look further right now, will update why later.

**Comment by sebastienros** [comment]
Our last merge broke your PR.
Would you mind fixing it?

**Comment by lukaskabrt** [comment]
I will update the PR tommorow

**Comment by lukaskabrt** [comment]
Conflict resolved, still need to move tenant URL generator to the correct place

**Comment by lukaskabrt** [comment]
Moved URL generator to the view. Everything should be now OK and ready for merge.

**Pull request merged**